processor/src/decoder/trace.rs

```diff
@@ -468,6 +468,23 @@ impl DecoderTrace {
         }
     }
 
+    /// Gets `n` values from the helper registers starting from the first hasher register that is
+    /// available for user operation at the current VM cycle.
+    ///
+    /// The specified USER_OP_HELPERS in the `hasher_trace` are used as helper registers, since they
+    /// are not required for hashing during execution of user operations.
+    pub fn get_user_op_helpers(&mut self, _cycle: usize, n: usize) -> Vec<Felt> {
+        let mut res = Vec::new();
+        for i in 0..n {
+            res.push(
+                *self.hasher_trace[USER_OP_HELPERS.start + i]
+                    .last()
+                    .expect("no last helper value"),
+            )
+        }
+        res
+    }
```

Comment:
A few comments:
- We usually name variables referring to VM cycle as `clk`.
- To make this work at any cycle, you should be able to index into the hasher trace registers like `self.hasher_trace[USER_OP_HELPERS.start + i][clk]`
- I think this function is used only for tests - so, let's add an attribute to make it available only in test context.
- I wonder if instead of returning `Vec<Felt>`, it should return `[Felt; NUM_USER_OP_HELPERS]`.